test/paymentmethods.collective.to.collective.test.js

```diff
@@ -0,0 +1,438 @@
+
+      // Setting up order
+      const order = {
+        fromCollective: { id: collective1.id },
+        collective: { id: collective2.id },
+        paymentMethod: { uuid: openCollectivePaymentMethod.uuid },
+        totalAmount: ocPaymentMethodBalance.amount,
+      }
+
+      // Executing queries
+      const res = await utils.graphqlQuery(createOrderQuery, { order }, user1);
+
+      // Then there should be no errors
+      res.errors && console.error(res.errors);
+      expect(res.errors).to.not.exist;
+
+      // Then Find Created Transaction
+      const orderFromCollective = res.data.createOrder.fromCollective;
+      const orderCollective = res.data.createOrder.collective;
+      const transaction = await models.Transaction.findOne({
+        where: { CollectiveId: orderCollective.id, amount: order.totalAmount }
+      });
+      // Then Check whether Created Transaction has NO fees
+      expect(transaction.FromCollectiveId).to.equal(orderFromCollective.id);
+      expect(orderCollective.id).to.equal(collective2.id);
+      expect(transaction.CollectiveId).to.equal(collective2.id);
+      expect(transaction.currency).to.equal(collective2.currency);
+      expect(transaction.platformFeeInHostCurrency).to.equal(0);
+      expect(transaction.hostFeeInHostCurrency).to.equal(0);
+      expect(transaction.paymentProcessorFeeInHostCurrency).to.equal(0);
+
+    });/** END OF "Collective 1 sends money to Collective 2 with no fees(same host)" */
+
+    it('Collective 1 fails to send money to Collective 3 because Collective 1\'s Host does not have a credit card', async () => {
+      // Add User 1 as Collective 1 Admin
+      await models.Member.create({
+        CreatedByUserId: user1.id,
+        MemberCollectiveId: user1.CollectiveId,
+        CollectiveId: collective1.id,
+        role: 'ADMIN'
+      });
+
+      //finding opencollective payment method for Collective 1
+      openCollectivePaymentMethod = await models.PaymentMethod.findOne({ where: {type: 'collective', CollectiveId: collective1.id}});
+
+      // get Balance given the created user
+      const ocPaymentMethodBalance = await openCollectivePaymentMethod.getBalanceForUser(user1);
+
+      // Setting up order
+      const order = {
+        fromCollective: { id: collective1.id },
+        collective: { id: collective3.id },
+        paymentMethod: { uuid: openCollectivePaymentMethod.uuid },
+        totalAmount: ocPaymentMethodBalance.amount/10,
+      }
+      // Executing queries
+      const res = await utils.graphqlQuery(createOrderQuery, { order }, user1);
+
+      // Then there should be Errors
+      expect(res.errors).to.exist;
+    });/** END OF "Collective 1 fails to send money to Collective 3 because Collective 1\'s Host does not have a credit card" */
+
+    it('Collective 1 fails to send money to Collective 3 because Collective 3\'s Host does not have a credit card', async () => {
```

Comment:
that shouldn't fail. The only requirement on the host of the recipient is to have a connected Stripe account (which they should as a host , but good to double check)